mantle/kola/harness.go

```diff
@@ -272,6 +273,74 @@ func testRequiresInternet(test *register.Test) bool {
 	return false
 }
 
+type DenyListObj struct {
+	Pattern string   `yaml:"pattern"`
+	Tracker string   `yaml:"tracker"`
+	Streams []string `yaml:"streams"`
+	Arches  []string `yaml:"arches"`
+}
+
+type ManifestData struct {
+	AddCommitMetadata struct {
+		FcosStream string `yaml:"fedora-coreos.stream"`
+	} `yaml:"add-commit-metadata"`
+}
+
+func parseDenyListYaml() error {
+	var objs []DenyListObj
+
+	// Parse kola-denylist into structs
+	denyListFile, err := ioutil.ReadFile("src/config/kola-denylist.yaml")
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	plog.Debug("Found kola-denylist.yaml. Processing listed denials.")
+	err = yaml.Unmarshal(denyListFile, &objs)
+	if err != nil {
+		return err
+	}
+
+	plog.Debug("Parsed kola-denylist.yaml")
+
+	// Get stream and arch
+	manifestFile, err := ioutil.ReadFile("src/config/manifest.yaml")
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	var manifest ManifestData
+	err = yaml.Unmarshal(manifestFile, &manifest)
+	if err != nil {
+		return err
+	}
+
+	stream := manifest.AddCommitMetadata.FcosStream
+	arch := system.RpmArch()
+
+	// Accumulate patterns filtering by stream and arch
+	plog.Debug("Processing denial patterns from yaml...")
+	for _, obj := range objs {
+		if !hasString(arch, obj.Arches) && len(obj.Arches) > 0 {
```

Comment:
I think concentually, this would read better the other way, i.e.

```suggestion
		if len(obj.Arches) > 0 && !hasString(arch, obj.Arches) {
```

and similarly below.

Reply:
agree

Reply:
:+1: